### PR TITLE
feat(IIdGenerator): Add IIdGenerator and extension to Object

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static MapperConfiguration Instance = new();
+
+    public MapperConfiguration() => _services = new ServiceCollection();
+
+    public void ConfigurationMapperOptions(IMapperOptionsExtension extension) => Extension = extension;
+
+    public Mapper CreateMapper()
+    {
+        Extension.AddService(_services);
```
Comment (doddgu) [inline src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs:20]: Why call AddService everytime?

File: src/Data/Masa.BuildingBlocks.Data.MappingExtensions/ObjectExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace System;
+
+public static class ObjectExtensions
+{
+    private static Mapper? _mapper;
+    private static Mapper GetMapper() => _mapper ??= MapperConfiguration.Instance.CreateMapper();
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data.MappingExtensions/ObjectExtensions.cs]: Modified to MapperFactory.Instance.CreateMapper

File: src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs]: There is no case of adding multiple times, it is only executed once during initialization, the following is the demo

initialization:

``` C#
MapperFactory.Instance.AddMapping().CreateMapper();
```

use:
``` C#
var request = new CreateUserRequest()
{
    Name = "Jim",
};
var user = request.Map<CreateUserRequest, User>();
Assert.IsNotNull(user);
Assert.IsTrue(user.Name == request.Name);
```

File: src/Data/Masa.BuildingBlocks.Data.Mapping/IMapperOptionsExtension.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Data.Mapping;
+
+public interface IMapperOptionsExtension
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data.Mapping/IMapperOptionsExtension.cs]: processed

File: src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data.Mapping/MapperConfiguration.cs]: It will indeed be executed once more in processing

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**